drivers/gpio/gpio_mcux_igpio.c

```diff
@@ -366,11 +371,14 @@ static const struct gpio_driver_api mcux_igpio_driver_api = {
 #define MCUX_IGPIO_PIN_DECLARE(n)						\
 	const struct pinctrl_soc_pinmux mcux_igpio_pinmux_##n[] = {		\
 		DT_FOREACH_PROP_ELEM(DT_DRV_INST(n), pinmux, PINMUX_INIT)	\
-	};
+	};									\
+	const uint8_t mcux_igpio_pin_gaps_##n[] =				\
+		DT_INST_PROP_OR(n, gpio_reserved_ranges, {});
```

Comment:
Can't you use the `struct gpio_pin_gaps` array type with static initialization here?

```suggestion
	const struct gpio_pin_gaps mcux_igpio_pin_gaps_##n[] =			\
		DT_INST_PROP_OR(n, gpio_reserved_ranges, {});
```

And have the `.gap_count` set to `ARRAY_SIZE(mcux_igpio_pin_gaps_##n)`

Reply:
Unless I'm missing something, I don't believe this will work. The issue is that when initializing a `struct gpio_pin_gaps` at compile time, the compiler would expect a definition like the following:
```
struct gpio_pin_gaps mcux_igpio_pin_gaps##n[] = {
   {
       3, /* start index */
       1, /* length */
   },
   {
       5, /* start index */
       4, /* length */
   },
}
```

However, the `DT_INST_PROP_OR` will expand to the following:
```
{3, 1, 4, 5}
```
This works fine to initialize an array of `uint8_t`, but for the `struct gpio_pin_gaps` it seems to generate a compiler warning: 

```
/home/danieldegrasse/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h:6854:59: warning: missing braces around initializer [-Wmissing-braces]
 6854 | #define DT_N_S_soc_S_gpio_400c0000_P_gpio_reserved_ranges {0 /* 0x0 */, 1 /* 0x1 */}
````

Reply:
Too bad, I was assuming it would've expanded to.

```c
{ {3, 1}, {4, 5} }
```